docroot/modules/custom/mass_entity_usage/tests/src/ExistingSite/EntityUsageTest.php

```diff
@@ -181,6 +181,96 @@ public function testEntityUsageTracking() {
     $this->assertUsageRows($node_org, 0);
   }
 
+  /**
+   * Usage count checks when updating and deleting referencing entities.
+   */
+  public function testUsageCountWhenEditingAndDeletingReferences() {
+    // Create node1.
+    $node_org = $this->createNode([
+      'type' => 'org_page',
+      'title' => 'Test Organization - Node 1',
+      'uid' => $this->user->id(),
+      'moderation_state' => MassModeration::PUBLISHED,
+    ]);
+    $this->processEntityUsageQueues();
+    // Check usages = 0.
+    $this->assertUsageRows($node_org, 0);
+
+    // Create node2 linking node1.
+    $node_curated_list_1 = $this->createNode([
+      'type' => 'curated_list',
+      'title' => 'Test Curated List - Node 2',
+      'uid' => $this->user->id(),
+      'moderation_state' => MassModeration::PUBLISHED,
+      'field_curatedlist_overview' => [
+        'value' => $node_org->toLink()->toString(),
+        'format' => 'basic_html',
+      ],
+    ]);
+    $this->processEntityUsageQueues();
+    // Usage for node1 should be 1.
+    $this->assertUsageRows($node_org, 1);
+
+    // Create node3, linking node1.
+    $node_curated_list_2 = $this->createNode([
+      'type' => 'curated_list',
+      'title' => 'Test Curated List - Node 3',
+      'uid' => $this->user->id(),
+      'moderation_state' => MassModeration::PUBLISHED,
+      'field_curatedlist_overview' => [
+        'value' => $node_org->toLink()->toString(),
+        'format' => 'basic_html',
+      ],
+    ]);
+    $this->processEntityUsageQueues();
+    // Usage for node1 should be 2.
+    $this->assertUsageRows($node_org, 2);
+
+    // Remove link from Node3 to Node1.
+    $node_curated_list_2->field_curatedlist_overview->setValue(NULL);
+    $node_curated_list_2->save();
+    $this->processEntityUsageQueues();
+    // Usage for node1 should be 1.
+    $this->assertUsageRows($node_org, 1);
+
+    // Delete node2.
+    $node_curated_list_1->delete();
+    $this->processEntityUsageQueues();
+    // Usage for node1 should be 0.
+    $this->assertUsageRows($node_org, 0);
+  }
+
+  /**
+   * Tests multiple references from one entity to another entity.
+   */
+  public function testCuratedList() {
```

Comment:
@deviantintegral after trying to replicate the issues you had, I ended up writing a test that is hopefully very similar to what you did. The test is showing the expected results.